### ISSUE TITLE
`"ipv4"` format validator rejects 0.x.y.z IPs

### ISSUE THREAD

**Issue by GREsau** [Stranger6667/jsonschema-rs#512]
```rust
let schema = jsonschema::compile(&json!({
    "type": "string",
    "format": "ipv4"
})).unwrap();

println!("{}", schema.is_valid(&json!("0.0.0.0")));
```

This outputs `false`, when I would expect `true` because 0.0.0.0 is a valid dotted-quad representation of an IPv4 address. All other IP addresses starting with 0 (e.g. 0.1.2.3) are similarly rejected.

This appears to be due to the fix for https://github.com/json-schema-org/JSON-Schema-Test-Suite/pull/469 (commit 7d32eff75f1af9deda610b2fb4ce5482ac80072d), but I think [this particular fix](https://github.com/Stranger6667/jsonschema-rs/blob/426df9c7c157899dd7f5ed3af96bb6d2e7a9bb4c/crates/jsonschema/src/keywords/format.rs#L209) is simultaneously:
- too strict, e.g. it rejects 0.0.0.0
- too loose because it allows leading zeroes in the second/third/fourth octets, e.g. it allows 1.0123.1.1
- unnecessary, because `Ipv4Addr`'s `FromStr` implementation already disallows leading zeros in any octet. _Edit: this was only added in rust 1.58 (https://github.com/rust-lang/rust/pull/86984), so the explicit check in this crate wasn't unnecessary when it was first implemented!_

**Comment by Stranger6667** [comment]
Thank you! Indeed, you are right! I’ll fix it soon

**Comment by Stranger6667** [comment]
Fixed in `0.19.1`
